### PR TITLE
feat/bugfix: Add "cargo mode" button in Outfitter

### PR DESCRIPTION
**Pull request by petervdmeer**
**Bugfix/Feature:** This PR implements the cargo button as discussed in issue #6668

## Bugfix/Feature Details
This PR adds a button at the bottom of the ships-list in the Outfitter to switch to cargo mode.

The solution is not very elegant; the button only results in all ships getting de-selected, and the Outfitter interface is still defined in code as it was before this PR. This PR is more of a "quick fix", not a long-term improvement where UI elements get fully defined in the datafiles.

## UI Screenshots
The button not active:
cargo_inactive

The button active:
cargo_active

## Usage Examples
Just press the button in the Outfitter

## Testing Done
Pressed the button in the Outfitter. And selected some Outfits and some ships to see if the code transfers properly.

## Performance Impact
N/A (not performance critical code)

### CODE REVIEW

**Comment by DeBlister** [comment]
Seems good to me.  I think I had to go to discord to check if anybody had a solution when I first tried out the new outfitter system and it still doesn't really make sense so some change should be made.  This helps a little as you said but maybe make it "Cargo/Storage" or "Cargo & Storage".  Seems like it takes up its own vertical space and that wording should definitely still fit across in that case.

**Comment by petervdmeer** [comment]
Thanks for the feedback, but  as I mentioned ([here](https://github.com/petervdmeer/endless-sky/pull/9#issuecomment-1095468616)) the button in this PR only enables the "buy to cargo" mode. More changes will be needed to also enable a "buy to storage" mode.

**Comment by DeBlister** [comment]
That's fair, I closed my request.  These buttons will be adequate but it might be better if we just added a new hotkey to avoid the need to "buy-through" from a shop to storage.

**Review by tehhowch — CHANGES_REQUESTED** [review]
This button should not scroll with the ship list - it should be at the top of the right sidebar, always visible above the ship selector. In the same horizonal row you could add a button for storage / "the garage".

To deduplicate code, I suggest adding a method for rendering a horizontal row of ShopPanel buttons inside a given region (e.g. `DrawButtons(const Rectangle &bounds, const vector<pair<string, fn>> &buttons)` ) and using it for this new button and the existing buy/ sell / leave.

**Comment by samrocketman** [comment]
Why not preserve the old behavior?

* Outfits for sale checked
* Outfits in cargo unchecked
* (new) Outfits stored on planet unchecked.

By the user checking the cargo box the buy/install button turns into cargo/purchase into cargo.

**Comment by petervdmeer** [comment]
> By the user checking the cargo box the buy/install button turns into cargo/purchase into cargo.

From an UI design perspective it feels more appropriate to me if the show/hide buttons just do 1 thing: show and hide the things that correspond to the checked box.

**Comment by petervdmeer** [comment]
> This button should not scroll with the ship list - it should be at the top of the right sidebar, always visible above the ship selector. In the same horizonal row you could add a button for storage / "the garage".

Yes, for the button as implemented now I can agree, but I was hoping that some proper artwork artist would be inspired to make 2-tile wide "buttons/clickzones" that would actually fit well at the location where I now placed the button. A photoshopped example (with not fully proper artwork):
buttons_artwork

If we would have such 2-tile wide zones/buttons, then it seems natural to place them at the end of the scroll-list, since players will typically first want to deal with their ships before they want to deal with cargo and/or storage.

File: source/ShopPanel.h [hunk 1]
```diff
@@ -52,6 +52,7 @@ class ShopPanel : public Panel {
 	// These are for the individual shop panels to override.
 	virtual int TileSize() const = 0;
 	virtual int VisiblityCheckboxesSize() const;
+	virtual std::string ShipsDeselectText() const;
```
[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/ShopPanel.h:55]: I was afraid that the original would go out of scope (become invalid) when the function returned.

**Comment by tehhowch** [comment]
Putting the buttons in the list of ships only works if they're to be used exactly like ships. I imagine the more common scenario is to want to toggle between certain ships and these two options, possibly repeatedly per session. Scrolling from the ship to select cargo, then scrolling back to select a different ship next to it, and so on, isn't quite great for UX when we can have them visible at the same time.

**Comment by petervdmeer** [comment]
I expect the common scenario to be that a player is first performing all outfitting of the ships (of a small fleet of less than 36 ships, so the buttons are visible anyway), and deal with "moving the leftover outfits to cargo and storage" after being done outfitting all ships.
Outfitting ships can be done from cargo, from storage and from stock while the ship being outfitted is selected, so I don't see a need to switch between the modes during ship-outfitting itself.

**Comment by petervdmeer** [comment]
I indicated in the PR text that this PR is a "quickfix", but the old behavior for cargo mode got restored in #6718 in the meantime, so we don't need a quickfix anymore.

Closing this PR in favor of a future more structural solution.

**Pull request closed without merging**